File: app/src/main/java/me/jfenn/alarmio/data/preference/AboutPreferenceData.java
package me.jfenn.alarmio.data.preference;

import android.content.Context;

import me.jfenn.alarmio.R;
import me.jfenn.attribouter.Attribouter;

public class AboutPreferenceData extends CustomPreferenceData {

    public AboutPreferenceData() {
        super(R.string.title_about);
    }

    @Override
    public String getValueName(ViewHolder holder) {
        return null;
    }

    @Override
    public void onClick(ViewHolder holder) {
        Context context = holder.getContext();

        Attribouter attribouter = Attribouter.from(context);
        int githubAuthKey = context.getResources().getIdentifier("githubAuthKey", "string", context.getPackageName());
        if (githubAuthKey != 0)
            attribouter = attribouter.withGitHubToken(context.getString(githubAuthKey));

        attribouter.show();
    }
}

File: app/src/main/java/me/jfenn/alarmio/data/preference/AboutPreferenceData.kt
package me.jfenn.alarmio.data.preference

import me.jfenn.alarmio.BuildConfig
import me.jfenn.alarmio.R
import me.jfenn.attribouter.Attribouter

/**
 * A preference item that opens the application's about screen.
 */
class AboutPreferenceData : CustomPreferenceData(R.string.title_about) {

    override fun getValueName(holder: CustomPreferenceData.ViewHolder): String? = null

    override fun onClick(holder: CustomPreferenceData.ViewHolder) {
        Attribouter.from(holder.context)
                .withGitHubToken(BuildConfig.GITHUB_TOKEN)
                .show()
    }
}
